### DISCUSSION TITLE
Is there an easy way to open a dev container on a remote host?

### DISCUSSION THREAD

**Discussion by tfenster** [devcontainers/community#53] [Q&A] [answered]
I am happily using devcontainers on a remote host, which works very well for me. The only issue is the "flow" to get to the container. I have to:

1. Open VS Code, switch to the remote explorer and select "Remote" as source
2. Select the remote VM and open VS Code there
3. Once it opens, select "Dev Containers" as source
4. Select the devcontainer that I need

Next time when I e.g. want to open another devcontainer or come back on the next day, the same story starts again and because the remote explorer remembers that I had the "Dev Containers" source selected, I even need to switch back to "Remote" first and then once more back to "Dev Containers". I know that the devcontainer also shows up in the "recently used" list, but that clutters fast and I have a lot of similar names, so even searching doesn't work too well for me. Is there any better way or shortcut that I could use to go from my local VS Code instance to being connected to the remote devcontainer?

**Comment by bamurtaugh** [comment, marked as answer]
Thanks for sharing this. In your list of recently opened files on the VS Code welcome page or from the command palette, you should be able to see and connect to your previous remote. Does that help?

**Reply by tfenster** [reply]
@bamurtaugh because of the number of different remote hosts and containers (with similar names) also not great, although certainly an improvement. I was just wondering, could it be an option to also show containers on a remote hosts, same opened workspaces / folders are shown:

![image](https://github.com/devcontainers/community/assets/15346084/5b3e4ff5-5c1d-426c-b367-83d97b0e3bf4)

**Reply by bamurtaugh** [reply]
Thank you for clarifying! I think this would be a feature request for the Dev Containers extension, which you could file here: https://github.com/microsoft/vscode-remote-release/issues.
